instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/integration/sessionreplay/SrTest.kt

```diff
@@ -23,13 +26,26 @@ import com.google.gson.internal.LazilyParsedNumber
 import okio.Buffer
 import org.assertj.core.api.Assertions.assertThat
 import org.assertj.core.api.recursive.comparison.RecursiveComparisonConfiguration
+import org.junit.After
 import java.io.ByteArrayOutputStream
 import java.io.File
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 import java.util.zip.Inflater
 
 internal abstract class SrTest<R : Activity> {
+
+    @After
+    fun tearDown() {
+        GlobalRumMonitor.get().stopSession()
+        Datadog.stopInstance()
+        GlobalRumMonitor::class.java.getDeclaredMethod("reset").apply {
```

Comment:
why it is needed? If instance is stopped, then nothing will be written anyway.

I can see the functional gap here however, because probably if we register instance with the same name as before, monitor will be still the old one, since we don't clean global RUM monitor when we stop SDK instance, I will create a ticket for it.

Reply:
oh damn...I think this is where my flakiness is coming from. Did not know that, right now between 2 tests I have same RUM context sometimes, this might explain this ?

Reply:
yeah...for now I want to keep this here to make sure everything is cleaned but yes I will open a ticket for later.